server/src/gameMachine/index.ts

```diff
@@ -0,0 +1,303 @@
+            [action.player]: { rightNeighbor: null },
+          };
+        case "SetNeighbor": {
+          return {
+            ...state,
+            [action.player]: { rightNeighbor: action.newRightNeighbor },
+          };
+        }
+
+        // When kicking, extract from the circle
+        case "ExtractPlayerFromCircle":
+        case "KickPlayer": {
+          const { [action.player]: kickedPlayerNeighbor, ...nextState } = state;
+          const oprhanedNeighbor = toEntries(nextState).find(
+            ([_, value]) => value?.rightNeighbor === action.player,
+          );
+
+          if (oprhanedNeighbor) {
+            return {
+              ...nextState,
+              [oprhanedNeighbor[0]]: {
+                rightNeighbor: kickedPlayerNeighbor?.rightNeighbor ?? null,
+              },
+            };
+          } else {
+            return nextState;
+          }
+        }
+
+        case "PlacePlayerInCircle": {
+          const { newRightNeighbor } = action;
+          const newLeftNeighborEntry = toEntries(state).find(
+            ([_, neighbors]) => neighbors?.rightNeighbor === newRightNeighbor,
+          );
+
+          return {
+            ...state,
+            ...(newLeftNeighborEntry
+              ? { [newLeftNeighborEntry[0]]: { rightNeighbor: action.player } }
+              : {}),
+            [action.player]: { rightNeighbor: newRightNeighbor },
+          };
+        }
+        default:
+          return state;
+      }
+    },
+
+    deadPlayers: (state = {}, action) => {
+      switch (action.type) {
+        case "RevivePlayer":
+        case "AddPlayer":
+          return {
+            ...state,
+            [action.player]: false,
+          };
+        case "KillPlayer":
+          return {
+            ...state,
+            [action.player]: true,
+          };
+        case "KickPlayer":
+          return removeKey(state, action.player);
+        default:
+          return state;
+      }
+    },
+    travelers: (state = {}, action) => {
+      switch (action.type) {
+        case "AddPlayer":
+          return {
+            ...state,
+            [action.player]: !!action.traveling,
+          };
+        case "KickPlayer":
+          return removeKey(state, action.player);
+        default:
+          return state;
+      }
+    },
+    alignmentsOverrides: (state = {}, action) => {
+      switch (action.type) {
+        case "AddPlayer":
+          return {
+            ...state,
+            // Should make this automatic if game started.
+            ...(action.traveling && { [action.player]: "Good" }),
+          };
+        case "KickPlayer":
+          return removeKey(state, action.player);
+        case "OverrideAlignment":
+          return {
+            ...state,
+            [action.player]: action.newAlignment,
+          };
+        default:
+          return state;
+      }
+    },
+
+    gmSecretHash: (state) => state ?? "t",
+    gameStatus: (state = "PlayersJoining", action) => {
```

Comment:
We should be careful that having just two statuses doesn't break anything in the interim we work on statuses